File: scratch/test5.cpp
#include <iostream>
#include <cmath>

int main() {
    float _zoom = 2.5f;
    float _earthRadius = 150.0f;
    float alt = 400.0f;
    float _cameraPitch = -45.0f;
    
    float visualAlt = alt;
    float r = _earthRadius + sqrtf(visualAlt) * 0.4f * _zoom;
    
    float _cameraFocusR = r; // assuming exact match
    
    float y = 0;
    float z = r;
    
    float pitchRad = _cameraPitch * 0.017453292519943295769236907684886f;
    
    float z_pitched = y * sinf(pitchRad) + z * cosf(pitchRad);
    float y_pitched = y * cosf(pitchRad) - z * sinf(pitchRad);
    
    float anchorBlend = (_zoom - 1.0f) / 1.5f;
    if (anchorBlend < 0.0f) anchorBlend = 0.0f;
    if (anchorBlend > 1.0f) anchorBlend = 1.0f;
    
    float focusOffset = _cameraFocusR * sinf(pitchRad) * anchorBlend;
    y_pitched += focusOffset;
    
    std::cout << "y_pitched: " << y_pitched << std::endl;
    std::cout << "z_pitched: " << z_pitched << std::endl;
    std::cout << "focusOffset: " << focusOffset << std::endl;
    return 0;
}

File: scratch/test5.py
import math

_zoom = 2.5
_earthRadius = 150.0
alt = 400.0
_cameraPitch = -45.0

visualAlt = min(alt, 20000.0)
r = _earthRadius + math.sqrt(visualAlt) * 0.4 * _zoom
_cameraFocusR = r

y = 0.0
z = r

pitchRad = _cameraPitch * 0.017453292519943295769236907684886

z_pitched = y * math.sin(pitchRad) + z * math.cos(pitchRad)
y_pitched = y * math.cos(pitchRad) - z * math.sin(pitchRad)

anchorBlend = max(0.0, min(1.0, (_zoom - 1.0) / 1.5))
focusOffset = _cameraFocusR * math.sin(pitchRad) * anchorBlend
y_pitched += focusOffset

print(f"y_pitched: {y_pitched}")
print(f"z_pitched: {z_pitched}")
print(f"focusOffset: {focusOffset}")
